primitiv/string_utils.h

```diff
@@ -28,6 +30,51 @@ std::string join(
   return ss.str();
 }
 
+/**
+ * Imprementation of std::to_string()
+ *
+ * Some libstdc++ (e.g. Android) do not support std::to_string().
+ * We support libraries that do not have std::to_string().
+ */
+
+#define DEF_TO_STRING(type, format, size) \
+  inline std::string to_string(type value) { \
+    char buffer[std::numeric_limits<type>::size]; \
```

Comment:
This array consumes a huge space of the stack (especially when converting from `long double`) that is usually not recommended. But using dynamic arrays (e.g., `vector`) requires some heap allocations. I have no good idea to fix this.